translate/storage/test_pypo.py

```diff
@@ -387,3 +387,10 @@ def test_mac_newlines(self):
         pofile = self.poparse(posource)
         assert len(pofile.units) == 1
         assert pofile.units[0].source == 'test me'
+
+    def test_mixed_newlines(self):
+        """checks that mixed newlines are properly parsed"""
+        posource = b'#Comment\n\n#: foo.c:124\r bar.c:124\r\nmsgid "test me"\nmsgstr ""\n'
```

Comment:
Please use triple quoted strings for holding the source in tests. It makes it easier to read, spot errors, fix them, and to copy and paste.